test_rfc6266_parser.py

```diff
@@ -56,42 +47,22 @@ def test_location_fallback():
     ).filename_unsafe == u'étoilé'
 
 
-def test_strict():
-    # Trailing ; means the header is rejected
-    assert parse_headers('attachment;').disposition == 'inline'
-    assert parse_headers('attachment; key=val;').disposition == 'inline'
-    try:
-        cd = parse_headers(
-            'attachment; filename="spa  ced";')
-    except ValueError:
-        assert True
-    else:
-        assert False, cd
-
-
 def test_relaxed():
-    assert parse_headers(
-        'attachment;', relaxed=True).disposition == 'attachment'
-    assert parse_headers(
-        'attachment; key=val;', relaxed=True).disposition == 'attachment'
-    cd = parse_headers(
-        'attachment; filename="spa  ced";',
-        relaxed=True)
+    assert parse_headers('attachment;').disposition == 'attachment'
+    assert parse_headers('attachment; key=val;').disposition == 'attachment'
+    cd = parse_headers('attachment; filename="spa  ced";')
     assert cd.filename_unsafe == u'spa ced'
 
 
-
-def test_roundtrip():
+@pytest.mark.parametrize("name", [
+    'a b', 'a b ', ' a b', 'a\"b', u'aéio o♥u"qfsdf!'
+])
+def test_roundtrip(name):
     def roundtrip(filename):
-        return parse_headers(build_header(filename)).filename_unsafe
-
-    def assert_roundtrip(filename):
-        assert roundtrip(filename) == filename
+        header = build_header(filename)
+        if sys.version_info >= (3,):
+            header = header.decode()
```

Comment:
There is shouldn't be such branching, types should be consistent over different versions.